File: ContactList/app/src/main/java/com/bergamin/contactlist/WebClient.java
package com.bergamin.contactlist;

import java.io.IOException;
import java.io.PrintStream;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.Scanner;

/**
 * Created by Guilherme on 30/10/2016.
 */

public class WebClient {
    public String post(String json){
        try {
            URL url = new URL("serverPath");
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setRequestProperty("Content-type","application/json");
            connection.setRequestProperty("Accept","application/json");
            connection.setDoOutput(true);
            PrintStream output = new PrintStream(connection.getOutputStream());
            output.println(json);
            connection.connect();
            Scanner scanner = new Scanner(connection.getInputStream());
            return scanner.next();
        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: Kotlin/ContactList/app/src/main/java/com/bergamin/contactlist/util/WebClient.kt
package com.bergamin.contactlist.util

import java.io.IOException
import java.io.PrintStream
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL
import java.util.*

/**
 * Created by Guilherme Taffarel Bergamin on 29/11/2017.
 */
class WebClient {
    fun post(json: String): String? {
        try {
            val serverPath = "Fill in this String with your own server path"
            val connection = URL(serverPath).openConnection() as HttpURLConnection

            connection.setRequestProperty("Content-type", "application/json")
            connection.setRequestProperty("Accept", "application/json")
            connection.doOutput = true

            PrintStream(connection.outputStream).println(json)
            connection.connect()

            return Scanner(connection.inputStream).next()
        }catch(e: MalformedURLException){
            e.printStackTrace()
        }catch(e: IOException){
            e.printStackTrace()
        }
        return null
    }
}
